.github/workflows/publish_release.yml

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Publish Release
+
+on:
+  push:
+    branches:
+      - "!**"
+    tags:
+    # Trigger workflow when a tag whose name matches the pattern
+    # "apache-arrow-{MAJOR}.{MINOR}.{PATCH}" is pushed.
+      - "*"
+      - "!*-rc*"
+
+permissions:
+  contents: write
+
+env:
+  GH_TOKEN: ${{ github.token }}
+
+jobs:
+  publish-release:
+    name: Publish Release 
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    if: github.ref_type == 'tag'
+    steps: 
+      - name: Checkout Arrow
+        uses: actions/checkout@v4
+        with:
+          path: arrow
+          fetch-depth: 0
+      - name: Get Latest Release Candidate's Git Tag
+        run: |
+          rc_tag=$(gh release list --repo ${{github.repository}} | \
+                    cut -f3 | \
+                    grep -F "${GITHUB_REF_NAME}-rc" | \
+                    head -n1)
+          echo "Latest Release Candidate Tag: ${rc_tag}"
+          echo "ARROW_RELEASE_CANDIDATE_TAG=${rc_tag}" >> ${GITHUB_ENV} 
+      - name: Create Release Title
+        run: |
+          title=$(gh release view ${ARROW_RELEASE_CANDIDATE_TAG} --repo ${{github.repository}} --json name --jq .name | \
+                      sed -E -e 's/ RC[0-9]+$//')
+          echo "Release Title: ${title}"
+          echo "ARROW_RELEASE_TITLE=${title}" >> ${GITHUB_ENV} 
+      - name: Create Release Notes URL
+        run: |
+          version=${GITHUB_REF_NAME#apache-arrow-}
+          release_note_url="https://arrow.apache.org/release/${version}.html"
+          echo "Release Notes URL: ${release_note_url}"
+          echo "ARROW_RELEASE_NOTES_URL=${release_note_url}" >> ${GITHUB_ENV} 
+      - name: Create GitHub Release
+        run: |
+          gh release create ${GITHUB_REF_NAME} \
+            --title "${ARROW_RELEASE_TITLE}" \
+            --verify-tag \
+            --notes "Release Notes: ${ARROW_RELEASE_NOTES_URL}" \
+            --repo ${{github.repository}}
```

Comment:
Could you also copy artifacts in the RC like https://github.com/apache/arrow-flight-sql-postgresql/blob/6e013be8491305295c76632b32f21b84296e5bb0/.github/workflows/release.yaml#L48-L53 ?

(Or we can do it when we work on #40956.)

Reply:
I originally did include a step to download the artifacts from RC candidate. However, [gh release download errors](https://github.com/mathworks/arrow/actions/runs/8621826934/job/23631558469#step:3:32) if there aren't any assets to download. I planned on adding this step in  #40956 since there'll be at least one asset to download at that point.

Reply:
Marking as resolved because I will add this step in #40956